src/jit/simdcodegenxarch.cpp

```diff
@@ -619,8 +619,8 @@ CodeGen::genSIMDIntrinsicInit(GenTreeSIMD* simdNode)
             // data section using movss/sd.  Similarly if op1 is a memory op we
             // would have loaded it using movss/sd.  Movss/sd when loading a xmm reg
             // from memory would zero-out upper bits. In these cases we can
-            // avoid explicitly zero'ing out targetReg.
-            bool zeroInitRequired = !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
+            // avoid explicitly zero'ing out targetReg if targetReg and op1Reg are the same.
+            bool zeroInitRequired = (targetReg != op1Reg) || !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
```

Comment:
I believe that Eugene is correct. It may be that this method could be enhanced to account for additional cases (or additional method(s) could be added). But the existing method expects that either 1) the targetReg will be unmodified except for the scalar (low) bits in the non-zero-init case, or 2) the targetReg will be cleared other than the scalar (low) bits in the zero-init case. It may also be the case that better naming and/or commenting is in order here (mea culpa).